### PR TITLE
autocorrecting Style/TrailingComma

### PR DESCRIPTION
**Pull request by lamont-granquist**
'chefstyle -a' fixed 1044 occurrences

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@chef/client-core add a whole lotta trailing commas

**Comment by danielsdeleo** [comment]
Is it possible to enforce this only for multi-line arrays/hashes? After seeing the effect, I'm inclined to not care about trailing commas for single line stuff, if we can pull it off. :+1: if not.

**Comment by lamont-granquist** [comment]
it actually does it only for multi-line.

the problem is that this is multi-line:

```ruby
         sip_paths= [
            '/System', '/bin', '/sbin', '/usr',
          ]
```

that could get fixed to either:

```ruby
         sip_paths= [ '/System', '/bin', '/sbin', '/usr' ]
```

or:

```
         sip_paths= [
            '/System',
            '/bin',
            '/sbin',
            '/usr',
          ]
```

but rubocop can't autocorrect those.

File: lib/chef/chef_fs/data_handler/acl_data_handler.rb [hunk 1]
```diff
@@ -11,8 +11,8 @@ def normalize(acl, entry)
             'read' => {},
             'update' => {},
             'delete' => {},
-            'grant' => {}
-            })
+            'grant' => {},
+            },)
```
Comment (lamont-granquist) [inline lib/chef/chef_fs/data_handler/acl_data_handler.rb]: ^ there's another edge case here which is less than delightful, but i believe there's another style cop which flags this as a bad way to match up the trailing paren, we just haven't gotten there yet.

Comment (lamont-granquist) [inline lib/chef/chef_fs/data_handler/acl_data_handler.rb]: actually it doesn't and that is mildly gross...

**Comment by danielsdeleo** [comment]
Ok, I see now. Carry on, then. :+1:

File: spec/functional/resource/execute_spec.rb [hunk 2]
```diff
@@ -89,7 +89,7 @@
       resource.environment({
         "SAWS_SECRET"  => "supersecret",
         "SAWS_KEY"     => "qwerty",
-      })
+      },)
```
Comment (lamont-granquist) [inline spec/functional/resource/execute_spec.rb]: so i think this one gets cleaned up later by dropping the braces entirely.

File: lib/chef/chef_fs/data_handler/acl_data_handler.rb (hunk 1 above)
Comment (lamont-granquist) [inline lib/chef/chef_fs/data_handler/acl_data_handler.rb]: the `{}` in the function call are actually optional here as well...  so all these get cleaned up later...

**Comment by thommay** [comment]
:+1:

**Pull request merged**